userspace/libsinsp/sinsp_filtercheck_event.cpp

```diff
@@ -650,10 +657,15 @@ void sinsp_filter_check_event::validate_filter_value(const char* str, uint32_t l
 	}
 }
 
-uint8_t* extract_argraw(sinsp_evt* evt, uint32_t* len, const char* argname) {
-	const sinsp_evt_param* pi = evt->get_param_by_name(argname);
+uint8_t* sinsp_filter_check_event::extract_argraw(sinsp_evt* evt,
+                                                  uint32_t* len,
+                                                  const char* argname) {
+	auto pi = evt->get_param_by_name(argname);
 
 	if(pi != NULL) {
+		m_arginfo = pi->get_info();
+		m_customfield.m_type = m_arginfo->type;
```

Comment:
Store correct field type and print format when we fetch them from the being-extracted event.